src/main/java/uk/gov/hmcts/ccd/domain/service/processor/DateTimeFormatParser.java

```diff
@@ -1,46 +1,36 @@
 package uk.gov.hmcts.ccd.domain.service.processor;
 
+import java.text.ParsePosition;
 import java.time.LocalDate;
 import java.time.LocalDateTime;
 import java.time.format.DateTimeFormatter;
 import java.time.format.DateTimeFormatterBuilder;
-import java.time.temporal.*;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.ChronoField;
+import java.time.temporal.TemporalAccessor;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
 
 import lombok.extern.slf4j.Slf4j;
-import org.joda.time.format.*;
 import org.springframework.stereotype.Component;
 
 @Component
 @Slf4j
 public class DateTimeFormatParser {
 
-    static final DateTimeFormatter DATE_TIME_FORMAT = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSS");
-    static final DateTimeFormatter DATE_TIME_FORMAT_MILLISECOND = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SS");
-    static final DateTimeFormatter DATE_FORMAT = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+    static final String DATE_TIME_FORMAT = "yyyy-MM-dd'T'HH:mm:ss.SSS";
+    static final String DATE_FORMAT = "yyyy-MM-dd";
     private static final int DEFAULT_YEAR = 1970;
 
     public String convertDateTimeToIso8601(String dateTimeFormat, String value) {
-        LocalDateTime dateTime;
-        try {
-            DateTimeFormatter inputFormat = (dateTimeFormat == null) ? DATE_TIME_FORMAT : getDateTimeFormatter(dateTimeFormat);
-            dateTime = LocalDateTime.parse(value, inputFormat);
-        } catch (Exception e) {
-            log.warn("Failed to parse following dateTime value {} with format {} attempting to parse with {}", value, dateTimeFormat, DATE_TIME_FORMAT);
-            dateTime = LocalDateTime.parse(value, DATE_TIME_FORMAT);
-        }
-        return dateTime.format(DATE_TIME_FORMAT);
+        return convert(Arrays.asList(dateTimeFormat, DATE_TIME_FORMAT), DATE_TIME_FORMAT, value);
     }
 
     public String convertDateToIso8601(String dateFormat, String value) {
-        LocalDate date;
-        try {
-            DateTimeFormatter inputFormat = (dateFormat == null) ? DATE_FORMAT : getDateFormatter(dateFormat);
-            date = LocalDate.parse(value, inputFormat);
-        } catch (Exception e) {
-            log.warn("Failed to parse following date value {} with format {} attempting to parse with {}", value, dateFormat, DATE_FORMAT);
-            date = LocalDate.parse(value, DATE_FORMAT);
-        }
-        return date.format(DATE_FORMAT);
```

Comment:
Please add check for **null** value parameter as well and add test cases accordingly